src/main/java/jep/SubInterpreterOptions.java

```diff
@@ -0,0 +1,195 @@
+ *     distribution.
+ */
+package jep;
+
+/**
+ * A configuration object containing additional options for constructing a SubInterpreter.
+ *
+ * These options map directly to the options documented at
+ * https://docs.python.org/3/c-api/init.html#c.PyInterpreterConfig.
+ * 
+ * These options are ignored if a Python version earlier than 3.12 is used with Jep.
+ *
+ * @since 4.2
+ */
+public class SubInterpreterOptions {
+
+    protected boolean isolated;
+
+    protected int useMainObmalloc = -1;
+
+    protected int allowFork = -1;
+
+    protected int allowExec = -1;
+
+    protected int allowThreads = -1;
+
+    protected int allowDaemonThreads = -1;
+
+    protected int checkMultiInterpeExtensions = -1;
+
+    protected int ownGIL = -1;
+
+    protected SubInterpreterOptions(boolean isolated){
+        this.isolated = isolated;
+    }
+
+    /**
+     * If this is false then the sub-interpreter will use its own "object"
+     * allocator state. Otherwise it will use (share) the main interpreter’s.
+     * 
+     * If this is false then check_multi_interp_extensions must be true. If this
+     * is true then gil must not be true.
+     *
+     * Shared Modules cannot be used when this is set to true.
+     *
+     * @param useMainObmalloc
+     *            whether the sub-interpreter will share the main interpreter's
+     *            allocator state.
+     * @return a reference to this SubInterpreterOptions
+     *
+     */
+    public SubInterpreterOptions setUseMainObmalloc(boolean useMainObmalloc) {
+        this.useMainObmalloc = useMainObmalloc ? 1 : 0;
+        return this;
+    }
+
+    /**
+     * If this is 0 then the runtime will not support forking the process in
+     * any thread where the sub-interpreter is currently active. Otherwise
+     * fork is unrestricted.
+     *
+     * Note that the subprocess module still works when fork is disallowed.
+     *
+     * @param allowFork
+     *            whether the  sub-interpreter will allow fork.
+     * @return a reference to this SubInterpreterOptions
+     *
+     */
+    public SubInterpreterOptions setAllowFork(boolean allowFork) {
+        this.allowFork = allowFork ? 1 : 0;
+        return this;
+    }
+
+    /**
+     * If this is false then the runtime will not support replacing the current
+     * process via exec (e.g. os.execv()) in any thread where the
+     * sub-interpreter is currently active. Otherwise exec is unrestricted.
+     *
+     * Note that the subprocess module still works when exec is disallowed.
+     *
+     * @param allowExec
+     *            whether the  sub-interpreter will allow exec.
+     * @return a reference to this SubInterpreterOptions
+     *
+     */
+    public SubInterpreterOptions setAllowExec(boolean allowExec) {
+        this.allowExec = allowExec ? 1 : 0;
+        return this;
+    }
+
+    /**
+     * If this is false then the sub-interpreter’s threading module won’t
+     * create threads. Otherwise threads are allowed.
```

Comment:
"won't create threads" would sound a bit better as "won't be able to create threads".

Reply:
Fixed in 4f2db10